pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"time"
+)
+
+// ServiceCAMetrics implements metrics gathering for this component.
+type ServiceCAMetrics struct {
+	expiryTime prometheus.Gauge
+}
+
+func NewServiceCAMetrics() *ServiceCAMetrics {
+	return &ServiceCAMetrics{
+		expiryTime: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "service_ca_expiry_time_seconds",
+			Help: "Reports the expiry time of service CA used in the cluster.",
+		}),
+	}
+}
+
+func (s *ServiceCAMetrics) Register() {
+	legacyregistry.MustRegister(s)
```

Comment:
Just register the metric directly then ServiceCAMetrics doesn’t need to implement the collector interface unnecessarily.

Reply:
@brancz Thanks for the tip. I already tried registering the metric directly `prometheus.MustRegister` but it did not work. I could not find the metric in web console when I queried for it. It is possible that I might be doing something wrong and I did not debug it further. I will take a look again.